x-pack/packages/ml/query_utils/src/default_query.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isFilterBasedSimpleQuery } from './filter_based_simple_query';
+import { isMatchAllQuery } from './match_all_query';
+import { isSimpleQuery } from './simple_query';
+import type { SearchQueryVariant } from './types';
+
+/**
+ * Checks if the provided query is a default query. A default query is considered as one that matches all documents,
+ * either directly through a `match_all` query, a `SimpleQuery` with a wildcard query string, or a `FilterBasedSimpleQuery`
+ * that contains a filter with a wildcard query or a `match_all` condition.
+ *
+ * @param {SearchQueryVariant} query - The query to check.
+ * @returns {boolean} True if the query is a default query, false otherwise.
+ */
+export function isDefaultQuery(query: SearchQueryVariant): boolean {
+  return (
+    isMatchAllQuery(query) ||
+    (isSimpleQuery(query) && query.query_string.query === '*') ||
```

Comment:
we could add a `isDefaultSimplyQuery` function to check this `*` value

Reply:
Good idea, updated in [9a6c5c7](https://github.com/elastic/kibana/pull/176534/commits/9a6c5c71b6f382577c5892b51e527eceed519ccd).